### PR TITLE
OCPBUGS-5066: fix profile updates in latency tests

### PR DESCRIPTION
**Pull request by mrniranjan**
backup profile and use initial backup to revert
the profile after latency tests are run.

Signed-off-by: Niranjan M.R <mrniranjan@redhat.com>

### CODE REVIEW

File: test/e2e/performanceprofile/functests/utils/profilesupdate/profile_update.go [hunk 1]
```diff
@@ -2,39 +2,37 @@ package profilesupdate
...
-func UpdateIsolatedReservedCpus(isolatedSet performancev2.CPUSet, reservedSet performancev2.CPUSet) error {
-	profile, err := profiles.GetByNodeLabels(testutils.NodeSelectorLabels)
-	if err != nil {
-		return fmt.Errorf("could not get the performance profile: %v", err)
-	}
-	updatedProfile := profile.DeepCopy()
-	updatedProfile.Spec.CPU = &performancev2.CPU{
+func UpdateIsolatedReservedCpus(profile *performancev2.PerformanceProfile, isolatedSet performancev2.CPUSet, reservedSet performancev2.CPUSet) error {
+
+	profile.Spec.CPU = &performancev2.CPU{
```
Comment (fromanirh) [inline test/e2e/performanceprofile/functests/utils/profilesupdate/profile_update.go:28]: this seems a step back though. Why is trusting the given profile more reliable than getting a fresh copy?

File: test/e2e/performanceprofile/functests/utils/profilesupdate/profile_update.go [hunk 2]
```diff
@@ -56,7 +54,15 @@ func ApplyProfile(profile *performancev2.PerformanceProfile) error {
...
 	testlog.Info("Applying changes in performance profile and waiting until mcp will start updating")
-	profiles.UpdateWithRetry(profile)
+	spec, err := json.Marshal(profile.Spec)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(testclient.Client.Patch(context.TODO(), profile,
+		client.RawPatch(
+			types.JSONPatchType,
+			[]byte(fmt.Sprintf(`[{ "op": "replace", "path": "/spec", "value": %s }]`, spec)),
+		),
+	)).ToNot(HaveOccurred())
```
Comment (fromanirh) [inline test/e2e/performanceprofile/functests/utils/profilesupdate/profile_update.go:64]: I'm not thrilled in adding expectations inside a utility method in this module. Actually I'm not thrilled about a patchwork approach, we should aim for some consistency.
But even before: why is Patch more reliable than Update?

File: test/e2e/performanceprofile/functests/utils/profilesupdate/profile_update.go (hunk 1 above)
Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/utils/profilesupdate/profile_update.go:28]: we are getting a fresh copy of the profile and passing that profile to this function.  We are doing this at:

https://github.com/openshift/cluster-node-tuning-operator/blob/master/test/e2e/performanceprofile/functests/5_latency_testing/5_latency_testing_suite_test.go#L51

File: test/e2e/performanceprofile/functests/utils/profilesupdate/profile_update.go (hunk 2 above)
Comment (mrniranjan) [reply test/e2e/performanceprofile/functests/utils/profilesupdate/profile_update.go:64]: I don't have answers now as to why patch works better than updateWithRetry except the issue is reproducible in this test case everytime. Also in the cases where we do use updateWithRetry , we have seen cases where tests do fail with profile updates fail .  I will investigate on this.  I can remove the exceptions and return err.

**Comment by mrniranjan** [comment]
closing this PR as the issue is not reproducible

**Pull request closed without merging**